### PR TITLE
create entities

### PR DESCRIPTION
**Pull request by dastanarzymatov**

### CODE REVIEW

File: src/main/java/kg/peaksoft/bilingualb6/entites/Client.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@Setter
+@NoArgsConstructor
+public class  Client {
+
+    @Id
+    @GeneratedValue(generator = "client_generator", strategy = GenerationType.SEQUENCE)
+    @SequenceGenerator(name = "client_generator", sequenceName = "client_id_sequence", allocationSize = 1)
+    private Long id;
+
+    private String full_name;
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Client.java:21]: Write full name with camel case (fullName)

File: src/main/java/kg/peaksoft/bilingualb6/entites/ClientsAnswer.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    private int numberOfWords;//user's answer's counter words
+
+    private int score;
+
+    @OneToOne(cascade = ALL)
+    private Content content;
+
+    @ManyToOne(cascade = {DETACH, MERGE, REFRESH},fetch = FetchType.LAZY)
+    private Question question;
+
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/ClientsAnswer.java:33]: Why is the relationship between the client answer and question ManyToOne?

File: src/main/java/kg/peaksoft/bilingualb6/entites/Option.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private Long id;
+
+    private String option;
+
+    private Boolean isTrue;
+
+    @ManyToOne(cascade = {MERGE, DETACH, PERSIST, REFRESH})
+    private Question question;
+
+    @ManyToOne(cascade = {DETACH,REFRESH,PERSIST,MERGE},fetch = FetchType.LAZY)
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Option.java:30]: I think the relationship between the option and the client answers can be uni directional

File: src/main/java/kg/peaksoft/bilingualb6/entites/Result.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    @OneToMany(cascade = CascadeType.ALL,mappedBy = "result")
+    private List<ClientsAnswer> clientsAnswers;
+
+    @ManyToOne(cascade = {MERGE, DETACH},fetch = FetchType.LAZY)
+    private Client client;
+
+    @ManyToOne(cascade = {MERGE,DETACH,REFRESH,PERSIST})
+    private Test test;
+}
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Result.java:41]: Here is the cascadeType PERSIST superfluous

File: src/main/java/kg/peaksoft/bilingualb6/entites/Question.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    @GeneratedValue(generator = "question_generator", strategy = GenerationType.SEQUENCE)
+    @SequenceGenerator(name = "question_generator", sequenceName = "question_id_sequence", allocationSize = 1)
+    private Long id;
+
+    private String title;
+
+    private String statement;
+
+    private String passage;
+
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Question.java:31]: add @Column(lenght = 10000)

File: src/main/java/kg/peaksoft/bilingualb6/entites/Question.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    @Id
+    @GeneratedValue(generator = "question_generator", strategy = GenerationType.SEQUENCE)
+    @SequenceGenerator(name = "question_generator", sequenceName = "question_id_sequence", allocationSize = 1)
+    private Long id;
+
+    private String title;
+
+    private String statement;
+
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Question.java:29]: add @column(lenght = 100000)

File: src/main/java/kg/peaksoft/bilingualb6/entites/Question.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    private String passage;
+
+    private Boolean isActive;
+
+    private int numberOfReplays;
+
+    private int duration;
+
+    private String shortDescription;
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Question.java:38]: add @column(lenght = 10000)

File: src/main/java/kg/peaksoft/bilingualb6/entites/Question.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+    private int minNumberOfWords;
+
+    private String correctAnswer;
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Question.java:42]: Isn't this field inside Content?

File: src/main/java/kg/peaksoft/bilingualb6/entites/Option.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+@Setter
+@NoArgsConstructor
+public class Option {
+
+    @Id
+    @GeneratedValue(generator = "option_generator", strategy = GenerationType.SEQUENCE)
+    @SequenceGenerator(name = "option_generator", sequenceName = "option_id_sequence", allocationSize = 1)
+    private Long id;
+
+    private String option;
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Option.java:23]: add @column(lenght = 10000)

File: src/main/java/kg/peaksoft/bilingualb6/entites/Content.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+public class Content {
+    @Id
+    @GeneratedValue(generator = "answer_generator", strategy = GenerationType.SEQUENCE)
+    @SequenceGenerator(name = "answer_generator", sequenceName = "answer_sequence", allocationSize = 1)
+    private Long id;
+
+    @Enumerated(EnumType.STRING)
+    private ContentType answerType;
+
+    private String content;
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Content.java:24]: add @column(lenght = 10000)

File: src/main/java/kg/peaksoft/bilingualb6/entites/Test.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+@Setter
+@NoArgsConstructor
+public class Test {
+
+    @Id
+    @GeneratedValue(generator = "test_generator", strategy = GenerationType.SEQUENCE)
+    @SequenceGenerator(name = "test_generator", sequenceName = "test_id_sequence", allocationSize = 1)
+    private Long id;
+
+    private String shortDescription;
```
Comment (Nurisaaa) [inline src/main/java/kg/peaksoft/bilingualb6/entites/Test.java:22]: add @column( lenght = 10000)

File: src/main/java/kg/peaksoft/bilingualb6/entites/Client.java (hunk 1 above)
Comment (ChyngyzIshen) [reply src/main/java/kg/peaksoft/bilingualb6/entites/Client.java:21]: As per new design we need add first name and last name

**Pull request merged**